static/schemas/2.0/user.2.0.json

```diff
@@ -0,0 +1,30 @@
+{
+    "title": "User.2.0",
+    "definitions": {
+        "PartialUser": {
+            "description": "A RackHD local user",
+            "type": "object",
+            "properties": {
+                "username": {
+                    "description": "The username of the user",
+                    "type": "string"
+                },
+                "password": {
+                    "description": "The password of the user",
+                    "type": "string"
+                },
+                "role": {
+                    "description": "The role of the user",
+                    "type": "string"
+                }
+            },
+            "additionalProperties": false
+        },
+        "User": {
+            "allOf": [
+                { "$ref": "#/definitions/PartialUser" },
+                { "required": [ "username", "password" ] }
```

Comment:
Not a big problem here, your code still works well here, but I suggest moving the `required` to within `PartialUser`, as they are strongly coupled with the PartialUser properties.